### PR TITLE
[NativeAOT-LLVM] Cleanup dotnet.js API integration with ExportsFile

### PR DESCRIPTION
**Pull request by maraf**
- `ExportsFile` is always set
- Use UTF8 without BOM when overriding exports file

### CODE REVIEW

**Comment by maraf** [comment]
@dotnet/nativeaot-llvm

File: src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
-    <EmccExportedFunction Include="_memset" />
-    <EmccExportedFunction Include="_ntohs" />
+    <EmccNativeExportedFunction Include="free" />
+    <EmccNativeExportedFunction Include="htons" />
+    <EmccNativeExportedFunction Include="main" />
+    <EmccNativeExportedFunction Include="malloc" />
+    <EmccNativeExportedFunction Include="memalign" />
+    <EmccNativeExportedFunction Include="memset" />
+    <EmccNativeExportedFunction Include="ntohs" />
+    <EmccExportedFunction Include="_%(EmccNativeExportedFunction)" />
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets]: Here's a snag I have not considered: `EmccExportedFunction` is a public API documented as being directly fed into Emscripten, i. e. you're expected to mangle it beforehand by adding the underscore to C symbols, but _not_ symbols defined in JS libraries. So it's hard to feed that into ILC as-is: no matter what you do, JS symbols will come out mangled on the other side.

In light of this, I think we have to keep the exports file rewriting for now (see e. g. https://github.com/emscripten-core/emscripten/issues/8380 for what the future might look like).

We could add another option to ILC that says "don't mangle symbols", but it doesn't seem worth it at this point in time.

BTW, there is a bit of code in the file under `Condition="'$(ExportsFile)' == ''` that could be deleted because the exports file is now always created so all of the exports can be routed through it.

File: src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets [hunk 2]
```diff
@@ -36,8 +38,15 @@
 
     <!-- Pseudo-namespace for the JS interop symbols. -->
     <DirectPInvoke Include="System.Runtime.InteropServices.JavaScript" />
+    <UnmanagedEntryPointsAssembly Include="System.Runtime.InteropServices.JavaScript" />
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets]: Unrelated change.
```suggestion
```

File: src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets (hunk 1 above)
Comment (maraf) [reply src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets]: Aye aye

**Comment by maraf** [comment]
@SingleAccretion Based on your comment, I have recycled the PR to fix the problem with UTF8-BOM

File: src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets [hunk 3]
```diff
@@ -76,20 +76,19 @@
       <!-- Exported functions required by user or runtime API -->
       <_DotNetJsLinkerFlag Include="-s DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=$(_EmccExportedLibraryFunction)" Condition="'$(_EmccExportedLibraryFunction)' != ''" />
       <_DotNetJsLinkerFlag Include="-s EXPORTED_RUNTIME_METHODS=$(_EmccExportedRuntimeMethods)" />
-      <_DotNetJsLinkerFlag Include="-s EXPORTED_FUNCTIONS=$(_EmccExportedFunctions)" Condition="'$(ExportsFile)' == ''" /><!-- Multiple args in the same rsp file is not support, add functions to app exports -->
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets:79]: `_EmccExportedFunctions` is now dead code.

**Review by SingleAccretion — APPROVED** [review]
LGTM modulo one comment and assuming green CI. Thank you!
